cloudinit/stages.py

```diff
@@ -342,9 +587,7 @@ def _get_datasources(self):
         cfg_list = self.cfg.get("datasource_list") or []
         return (cfg_list, pkg_list)
 
-    def _restore_from_checked_cache(self, existing):
-        if existing not in ("check", "trust"):
```

Comment:
Since `Init()` is annotated with the `stages.CacheMode`, and all callsites of `Init()` are checked by mypy, we shouldn't need to do these runtime checks anymore.